validation.go

```diff
@@ -210,7 +210,7 @@ func (v validator) Validate(config map[string]string) error {
 		}
 		multiErr = multierr.Append(multiErr, v.validateParamValue(pKey, cfg[pKey]))
 	}
-	return multiErr
+	return cfg, multiErr
```

Comment:
Nitpick: At least to me, it doesn't feel quite intuitive that a Validate method also returns a fully populated config. IMHO it would be nice if we either rename it, or just call two methods. 

Side question: does the validator check the default values too?